packages/abi-coder/src/index.ts

```diff
@@ -14,4 +14,3 @@ export {
   calculateVmTxMemory,
   ENCODING_V1,
 } from './utils/constants';
-export { AbiCoder } from './AbiCoder';
```

Comment:
Stopped exporting this as it's an internal detail. Those who want to work with abis should do so via the `Interface` class, e.g. the [changes](https://github.com/FuelLabs/fuels-ts/pull/2826/files#diff-e3d0c6c1264fe3529431371fe4d50cd6ef11a4b1baabbc774600053db5c764ccR59) I did to the `encode-and-decode` documentation.

Reply:
The `Interface` and `AbiCoder` work with the same data and both have the same rigidity (can't use custom coders, the `argument` must exist in the abi). The only difference is that `Interface` gives structure to ABI constructs like functions, logs, configurables. What we can do to allow users encoding of any type included in the abi on the `Interface` is we can add additional `encodeType` and `decodeType` methods. This would cover the use case you're mentioning.
```ts
// class Interface

/**
 * @param concreteTypeId - The id of the type you want to encode.
 * It must exist in the concreteTypes array of the abi given to the Interface.
 * @param data - The data you want to encode.
 * @param options - Encoding options
 * @returns Uint8Array of encoded data
 */
encodeType(concreteTypeId: string, data: InputValue, options: EncodingOptions) {
  return AbiCoder.encode(
    this.jsonAbi,
    this.resolvableMetadataTypes,
    concreteTypeId,
    data,
    options
  );
}

/**
 * @param concreteTypeId - The id of the type you want to decode.
 * It must exist in the concreteTypes array of the abi given to the Interface.
 * @param data - The data you want to decode.
 * @param options - Encoding options
 * @returns decoded data in the shape of the concrete type
 */
decodeType(concreteTypeId: string, data: Uint8Array, options: EncodingOptions) {
  return AbiCoder.decode(
    this.jsonAbi,
    this.resolvableMetadataTypes,
    concreteTypeId,
    data,
    0,
    options
  );
}
```